pkg/operator/ceph/pool/controller.go

```diff
@@ -288,3 +311,32 @@ func updateStatus(client client.Client, poolName types.NamespacedName, status st
 	}
 	logger.Debugf("pool %q status updated to %q", poolName, status)
 }
+
+func configureRBDStats(clusterContext *clusterd.Context, clusterName string) error {
+	cephBlockPoolList := cephv1.CephBlockPoolList{}
+	var enableStatsForPools []string
+	err := clusterContext.Client.List(context.TODO(), &cephBlockPoolList)
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve list of CephBlockPool")
+	}
+	for _, cephBlockPool := range cephBlockPoolList.Items {
+		if cephBlockPool.GetDeletionTimestamp() == nil && cephBlockPool.Spec.EnableRBDStats {
+			// list of CephBlockPool with enableRBDStats set to true and not marked for deletion
+			enableStatsForPools = append(enableStatsForPools, fmt.Sprintf("%s/%s", cephBlockPool.Name, cephBlockPool.Namespace))
+		}
+	}
+	logger.Debugf("RBD per-image IO statistics will be collected for %v", enableStatsForPools)
+	// using Ceph status to identify Active Mgr name
+	status, err := cephclient.Status(clusterContext, clusterName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve status")
+	}
+	mgrName := status.MgrMap.ActiveName
+	logger.Debugf("Active mgr is mgr.%s", mgrName)
```

Comment:
Instead of querying the active mgr, I believe you can just set the config to apply to all mgrs with the name `mgr.` (the dot is important) If rook did support multiple mgrs, I would expect the same config to apply to all mgrs.

Reply:
How do we retrieve list of all mgrs (for both converged and external cluster)?

I thought when there are multiple mgrs, there'd still be only one active (leader) or even if all are active there'd be a common config for all. Not sure how multiple mgrs with different config would reconcile the same cluster.

Reply:
@umangachapagain not sure to understand what you mean, I think @travisn is suggesting what we discussed earlier but you told me it was not possible after all. Please remind us why we can't just do `config set mgr <foo> <bar>`?

Reply:
Even the doc mentions it, https://docs.ceph.com/docs/master/mgr/prometheus/#rbd-io-statistics, am I missing something?

Reply:
I tired `mgr` instead of `mgr.a` and it didn't work on Nautilus but worked on Octopus.

So, if we want all mgrs to be configured on Nautilus, we'll have to figure out all IDs (from a to z) whichever is used and append to `mgr` and run the command for each. (I hope I am totally wrong about this and there is an easy way to configure all mgrs at once even if they are inactive)

Reply:
Ahhh Nautilus versus Octopus behavior, now I remember, that's very unfortunate. For external cluster, it won't be from a to z. On converged mode, we only have a single mgr and it will always have the same name. We can have a check for the Ceph version and apply the correct name.

Reply:
Instead of `mgr` does `mgr.` work on nautilus? We are using "mon." consistently for mons

Reply:
🤷‍♂️ , @umangachapagain? What's not working?

```
[root@mon0 ~]# ceph config set mgr mgr/prometheus/rbd_stats_pools test

[root@mon0 ~]# ceph -v
ceph version 14.2.9 (581f22da52345dba46ee232b73b990f06029a2a0) nautilus (stable)

[root@mon0 ~]# ceph osd dump|grep test
pool 11 'test' replicated size 1 min_size 1 crush_rule 1 object_hash rjenkins pg_num 8 pgp_num 8 autoscale_mode warn last_change 61 flags hashpspool stripe_width 0

[root@mon0 ~]# ceph config dump|grep mgr/prometheus/rbd_stats_pools
  mgr         advanced mgr/prometheus/rbd_stats_pools      test                                     *  
```

Am I missing something? Even the Nautilus branch of the doc recommends `mgr`.

Reply:
Here's what I observed
```

Reply:
Here's what I observed on
```
[root@rook-ceph-tools-7647684d74-f9vll /]# ceph version
ceph version 15.2.4 (7447c15c6ff58d7fce91843b705a268a1917325c) octopus (stable)
```
```
[root@rook-ceph-tools-7647684d74-f9vll /]# ceph config get mgr mgr/prometheus/rbd_stats_pools
replicapool
[root@rook-ceph-tools-7647684d74-f9vll /]# ceph config get mgr.a mgr/prometheus/rbd_stats_pools
device_health_metrics
[root@rook-ceph-tools-7647684d74-f9vll /]# ceph config get mgr. mgr/prometheus/rbd_stats_pools
replicapool
```

`mgr` and `mgr.` are the same but `mgr.a` is different.

Reply:
@pcuzner Maybe you can help us understand this?